cmake/modules/hunter_pkgconfig_export_target.cmake

```diff
@@ -0,0 +1,128 @@
+# ----------------------------------------------------------------------
+# Auto creation of CMake targets from pkg-config
+#
+# Copyright (C) 2017 Alexandre Pretyman. All rights reserved.
+#
+# Looks for a pkgconfig module of a given name and exports a cmake
+# target of the same name
+# ----------------------------------------------------------------------
+
+include(FindPkgConfig)
+
+include(hunter_internal_error)
+include(hunter_status_debug)
+
+function(hunter_pkgconfig_export_target PKG_CONFIG_MODULE)
+  if(TARGET "${PKG_CONFIG_MODULE}")
+    return()
+  endif()
+  pkg_check_modules(${PKG_CONFIG_MODULE} ${PKG_CONFIG_MODULE})
```

Comment:
Using cmake 3.8.0-rc2 with `IMPORTED_TARGET`, it did not link properly:
```
CMakeFiles/x264-example.dir/example.c.o: In function `main':
example.c:(.text+0xc8): undefined reference to `x264_param_default_preset'
example.c:(.text+0x11b): undefined reference to `x264_param_apply_profile'
example.c:(.text+0x144): undefined reference to `x264_picture_alloc'
example.c:(.text+0x15b): undefined reference to `x264_encoder_open_148'
example.c:(.text+0x281): undefined reference to `x264_encoder_encode'
example.c:(.text+0x312): undefined reference to `x264_encoder_encode'
example.c:(.text+0x369): undefined reference to `x264_encoder_delayed_frames'
example.c:(.text+0x380): undefined reference to `x264_encoder_close'
example.c:(.text+0x38f): undefined reference to `x264_picture_clean'
example.c:(.text+0x3af): undefined reference to `x264_encoder_close'
example.c:(.text+0x3c1): undefined reference to `x264_picture_clean'
collect2: error: ld returned 1 exit status
```

While using hunter version it worked fine.